msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java

```diff
@@ -15,74 +12,43 @@
 
 class OAuthAuthorizationGrant extends AbstractMsalAuthorizationGrant {
 
-    private AuthorizationGrant grant;
     private final Map<String, List<String>> params = new LinkedHashMap<>();
 
-    OAuthAuthorizationGrant(final AuthorizationGrant grant, Set<String> scopesSet, ClaimsRequest claims) {
-        this(grant, scopesSet != null ? String.join(" ", scopesSet) : null, claims);
-    }
-
-    String addCommonScopes(String scopes) {
-        Set<String> allScopes = new HashSet<>(
-                Arrays.asList(COMMON_SCOPES_PARAM.split(SCOPES_DELIMITER)));
-
-        if (!StringHelper.isBlank(scopes)) {
-            allScopes.addAll(Arrays.asList(scopes.split(SCOPES_DELIMITER)));
+    /**
+     * Constructor to create an OAuthAuthorizationGrant
+     *
+     * @param params parameters relevant for the specific authorization grant type
+     * @param scopes additional scopes which will be added to a default set of common scopes
+     * @param claims optional claims
+     */
+    OAuthAuthorizationGrant(Map<String, List<String>> params, Set<String> scopes, ClaimsRequest claims) {
+        this.scopes = new HashSet<>(AbstractMsalAuthorizationGrant.COMMON_SCOPES);
+
+        if (scopes != null) {
+            this.scopes.addAll(scopes);
         }
-        return String.join(SCOPES_DELIMITER, allScopes);
-    }
 
-    OAuthAuthorizationGrant(final AuthorizationGrant grant, String scopes, ClaimsRequest claims) {
-        this.grant = grant;
+        this.params.put(SCOPE_PARAM_NAME, Collections.singletonList(String.join(" ", this.scopes)));
 
-        String allScopes = addCommonScopes(scopes);
-        this.scopes = allScopes;
-        params.put(SCOPE_PARAM_NAME, Collections.singletonList(allScopes));
+        if (params != null) {
+            this.params.putAll(params);
+        }
 
         if (claims != null) {
             this.claims = claims;
-            params.put("claims", Collections.singletonList(claims.formatAsJSONString()));
-        }
-    }
-
-    OAuthAuthorizationGrant(AuthorizationGrant grant, String scopes, Map<String, List<String>> extraParams) {
-        this.grant = grant;
-
-        String allScopes = addCommonScopes(scopes);
-        this.scopes = allScopes;
-        this.params.put(SCOPE_PARAM_NAME, Collections.singletonList(allScopes));
-
-        if (extraParams != null) {
-            this.params.putAll(extraParams);
-        }
-    }
-
-    OAuthAuthorizationGrant(AuthorizationGrant grant, Map<String, List<String>> params) {
-        this.grant = grant;
-
-        if (params != null) {
-            this.params.putAll(params);
+            this.params.put("claims", Collections.singletonList(claims.formatAsJSONString()));
         }
     }
```

Comment:
It seems like nothing can modify values in this class after construction. This doesn't need to reconstruct the output map every time. Setup the extra values into the `params` field at construction.

Reply:
Good point, in the latest commit the "client_info" parameter is added to the map by the constructor, and the toParameters() method simply returns an unmodifiable version of the existing map.